mlflow/store/db_migrations/versions/3500859a5d39_add_model_aliases_table.py

```diff
@@ -0,0 +1,36 @@
+"""Add Model Aliases table
+
+Revision ID: 3500859a5d39
+Revises: 97727af70f4d
+Create Date: 2023-03-09 15:33:54.951736
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from mlflow.store.model_registry.dbmodels.models import SqlRegisteredModelAlias
+
+# revision identifiers, used by Alembic.
+revision = "3500859a5d39"
+down_revision = "97727af70f4d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        SqlRegisteredModelAlias.__tablename__,
+        sa.Column("name", sa.String(length=256), primary_key=True, nullable=False),
+        sa.Column("alias", sa.String(length=250), primary_key=True, nullable=False),
+        sa.Column("version", sa.Integer(), nullable=False),
+        sa.PrimaryKeyConstraint("name", "alias", name="registered_model_alias_pk"),
+        sa.ForeignKeyConstraint(
+            ("name", "version"),
+            ("model_versions.name", "model_versions.version"),
+            onupdate="cascade",
+            name="registered_model_alias_name_version_fkey",
+        ),
+    )
+
+
+def downgrade():
+    pass
```

Comment:
Did we want to have an actual downgrade command here?

```def downgrade():
  op.drop_constraint("registered_model_alias_name_version_fkey")
  op.drop_constraint("registered_model_alias_pk")
  op.drop_table(SqlRegisteredModelAlias.__tablename__)
  ```
  
  I know we don't typically put downgrade statements in table mutation statements, but would this be something that we'd want to support a rollback with? 
  If not, feel free to disregard this comment :)

Reply:
I was just wondering what would happen if:
- upgrade to create table
- downgrade without removing table
- upgrade to a later version that has this upgrade in the dep chain

Wouldn't the migration fail?
Something like this:

sqlalchemy.exc.InternalError: (InternalError) (1050, u"Table '<alias table>' already exists")

I actually was wondering the same thing as you about 'why isn't this defined elsewhere?' and it's probably the risk of dropping data from tables (although if a rollback happened, and then an upgrade... some poor DBA will have a lot of temporary table copying to do to re-populate that data before manually dropping the table and proceeding with the upgrade). 

Not a blocker from my perspective, though. Great work on this and excellent perseverance on struggling through the nuances of alembic! :D

Reply:
Sure let's start the trend then with this migration! And if we hear about data loss issues, we can address accordingly

Reply:
I'd rather we make the migration idempotent. Can we do that?

Reply:
Something like this probably :) @arpitjasa-db 
```python
from sqlalchemy.engine.reflection import Inspector

def get_existing_tables():
  connection = op.get_bind()
  inspector = Inspector.from_engine(connection)
  return inspector.get_table_names()

if SqlRegisteredModelAlias.__tablename__ not in get_existing_tables():
  op.create_table(......)
```

Reply:
Added! let's hope it works 😂

Reply:
ye of little faith ;)

Reply:
Can you blame me given this "journey"? 😂